lib/src/main/java/com/diffplug/spotless/npm/NpmFormatterStepStateBase.java

```diff
@@ -43,11 +42,7 @@
 
 	private static final long serialVersionUID = 1460749955865959948L;
 
-	@SuppressWarnings("unused")
-	private final FileSignature packageJsonSignature;
-
-	@SuppressFBWarnings("SE_TRANSIENT_FIELD_NOT_RESTORED")
-	public final transient File nodeModulesDir;
+	private final File buildDir;
```

Comment:
I think I said something like "state shouldn't write to disk", but I was wrong.  This change will break up-to-date checking and the build cache.  I'll revert it but keep your logging.

Reply:
Thanks for checking this in-depth!